sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,301 @@
+    assert planar_joint.parent == parent
+    assert planar_joint.child == child
+    assert planar_joint._parent_joint_location == Vector(0)
+    assert planar_joint._child_joint_location == l * child.frame.y
+    assert planar_joint.parent_axis == parent.frame.y
+
+    theta = dynamicsymbols('planar_joint_theta')  # rotation around z axis.
+    thetad = dynamicsymbols('planar_joint_theta', 1)
+    omega = dynamicsymbols('planar_joint_omega')
+    disx = dynamicsymbols('planar_joint_disx')  # translation along x axis.
+    disxd = dynamicsymbols('planar_joint_disx', 1)
+    velx = dynamicsymbols('planar_joint_velx')
+    disy = dynamicsymbols('planar_joint_disy')  # translation along y axis.
+    disyd = dynamicsymbols('planar_joint_disy', 1)
+    vely = dynamicsymbols('planar_joint_vely')
+
+    assert planar_joint.coordinates == [theta, disx, disy]
+    assert planar_joint.speeds == [omega, velx, vely]
+    assert planar_joint.kds == [thetad - omega, disxd - velx, disyd - vely]
+
+    child_joint_point = planar_joint.child_joint_point
+    parent_joint_point = planar_joint.parent_joint_point
+    assert parent_joint_point.name == 'planar_joint_parent_joint'
+    assert child_joint_point.name == 'planar_joint_child_joint'
+    assert child_joint_point.pos_from(parent_joint_point) == Vector(0)
+    assert child_joint_point.pos_from(child.masscenter) == l * child.frame.y
+    assert parent_joint_point.pos_from(parent.masscenter) == Vector(0)
+
+    vecx = cross(planar_joint.parent_axis, planar_joint._child_joint_location)
+    assert planar_joint.vecx == vecx
+    vecy = cross(planar_joint.parent_axis, vecx)
+    assert planar_joint.vecy == vecy
+    assert planar_joint.vecz == planar_joint.parent_axis
+
+    assert child.frame.ang_vel_in(parent.frame) == omega * planar_joint.parent_axis
+    assert child_joint_point.vel(parent.frame) == disx * vecx + disy * vecy
+
+
+def test_planar_joint_arguments():
+    parent = Body('parent')
+    child = Body('child')
+    l1 = Symbol('l1')
+    l2 = Symbol('l2')
+    planar_joint = PlanarJoint('planar_joint', parent, child,
+                               parent_point_pos=(0, l1, 0),
+                               child_point_pos=(0, l2, 0),
+                               parent_axis=parent.frame.y,
+                               child_axis=child.frame.y)
+    assert planar_joint.name == 'planar_joint'
+    assert planar_joint.parent == parent
+    assert planar_joint.child == child
+    assert planar_joint.parent_axis == parent.frame.y
+    assert planar_joint.child_axis == child.frame.y
+    assert planar_joint._parent_joint_location == l1 * parent.frame.y
+    assert planar_joint._child_joint_location == l2 * child.frame.y
+
+
+def test_spherical_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    spherical_joint = SphericalJoint('spherical_joint', parent, child,
+                                     child_point_pos=(0, l, 0))
```

Comment:
> Tell me if I am wrong, so what you mean is that we should give user an option to pass rotation order like xyz or yzy etc. because one rotation order xyz which is used now will not work for every problem.

That is correct.

> Secondly, the names of the variables should be according to convention. Right?

It would be nice to choose the rotation variables from the typical Euler parameter names.